lib/src/typewriter.dart

```diff
@@ -236,21 +231,23 @@ class _TypewriterState extends State<TypewriterAnimatedTextKit>
 
     if (mounted) setState(() {});
 
+    final textLen = _textCharacters[_index].length;
     _controller = AnimationController(
-      duration: _texts[_index]['speed'] * _texts[_index]['text'].length,
+      duration: widget.speed * textLen,
       vsync: this,
     );
 
-    _typewriterText =
-        StepTween(begin: 0, end: _texts[_index]['text'].length + 8)
-            .animate(_controller)
-              ..addStatusListener(_animationEndCallback);
+    _typewriterText = StepTween(
+      begin: 0,
+      end: textLen + 8,
```

Comment:
Does anybody know why 8 is being added?  What is significant about 8?

Reply:
`8` is added for the `_` typing animation after the given text is animated or typed.

Reply:
Why 8 and not 1?

Reply:
Idk it's a random number that I added at the time of development, a lot of code has been changed after that.